src/App.js

```diff
@@ -14,11 +15,75 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    listVisibility: false,
+  }
+
+  showList = () => {
+    this.setState(state => ({
+      ...state,
+      listVisibility: true,
+    }));
```

Comment:
```suggestion
    this.setState({listVisibility: true});
```
For this operation you don't need the previous state, so you can just change visibility without using functional setState